site.py

```diff
@@ -0,0 +1,84 @@
+
+
+app = Flask(__name__)
+
+# Confirm that a provided file has an acceptable extension.
+# In this case, '.csv' is the only allowed extension.
+def allowed_file(filename):
+    return ('.' in filename) and (filename.rsplit('.', 1)[1] in ALLOWED_EXTENSIONS)
+
+# Retrieve the files uploaded by the user, and if they are valid,
+#  store them in a new generated subfolder of tmp/.
+@app.route('/', methods=['GET', 'POST'])
+def upload_file():
+    if request.method == 'POST':
+        metadatafile = request.files['metadatafile']
+        datafile = request.files['datafile']
+        
+        if metadatafile and metadatafile.filename.endswith('.csv') \
+           and datafile and datafile.filename.endswith('.csv'):
+            print "Saving files..."
+            app.config['UPLOAD_FOLDER'] = 'tmp/'+str(random.randint(10000,99999))+'/'
+            while os.path.exists(app.config['UPLOAD_FOLDER']):
+                app.config['UPLOAD_FOLDER'] = 'tmp/'+str(random.randint(10000,99999))+'/'
+            
+            os.mkdir(app.config['UPLOAD_FOLDER'])
+            metadatafile.save(
+                os.path.join(
+                    app.config['UPLOAD_FOLDER'],
+                    metadatafile.filename
+                )
+            )
+            datafile.save(
+                os.path.join(
+                    app.config['UPLOAD_FOLDER'],
+                    datafile.filename
+                )
+            )
+            return redirect(url_for(
+                                'uploaded_file', 
+                                datafilename = datafile.filename, 
+                                metadatafilename = metadatafile.filename
+                            ))
+
+    # If method is not 'POST' (i.e., the user has not uploaded files yet)
+    #  or if files are not accepted, load the form.
+    return render_template('index.html')
+
+# Generate the Docker container from the uploaded files, 
+#  and redirect to the new Harvest instance.
+@app.route('/uploads/<datafilename>/<metadatafilename>')
+def uploaded_file(datafilename, metadatafilename):
+    print 'Files accepted. Generating Docker Container...' 
+    containerid = subprocess.check_output([
+                      'docker', 'run', '-d', '-P',
+                      '-e', ('DATA_FILE=' + datafilename), 
+                      '-e', ('METADATA_FILE=' + metadatafilename),
+                      '-v', (os.path.abspath(app.config['UPLOAD_FOLDER']) + ':/input'),
+                      'dbhi/redcap-harvest'
+                  ])
+    containerid = containerid[0:9]  # Get short container id for use in inspect
+    containerinfo = json.loads(subprocess.check_output(['docker','inspect',containerid]))
+   
+    print 'Waiting for harvest to load...' 
+    harvestIP = 'http://'+subprocess.check_output(['boot2docker', 'ip']).strip()+':'+\
+                  containerinfo[0]['NetworkSettings']['Ports']['8000/tcp'][0]['HostPort']
+    while True:
```

Comment:
Never assume something will work as you expect :) Change this unbounded while loop to:

```python
for _ in range(30):
    ...
```

Which will result in ~30 second timeout given the sleep function being used.